python/orca/tutorial/NCF/process_xshards.py

```diff
@@ -51,38 +53,86 @@ def ng_sampling(data, user_num, item_num):
 
     features_fill = features_ps + features_ng
     labels_fill = labels_ps + labels_ng
-    data_XY = pd.DataFrame(data=features_fill, columns=["user", "item"], dtype=np.int64)
+    data_XY = pd.DataFrame(data=features_fill, columns=clms, dtype=np.int64)
     data_XY["label"] = labels_fill
     data_XY["label"] = data_XY["label"].astype(np.float)
     return data_XY
 
 
-def split_dataset(data):
+def split_dataset(data, total_cols):
+    data = data.loc[:, total_cols]
     train_data, test_data = train_test_split(data, test_size=0.2, random_state=100)
     return train_data, test_data
 
 
-def prepare_data(dataset_dir):
-    data = read_csv(os.path.join(dataset_dir, 'ratings.dat'),
-                    sep="::", header=None, names=['user', 'item'],
-                    usecols=[0, 1], dtype={0: np.int64, 1: np.int64})
-    data = data.partition_by("user")
-
-    user_set = set(data["user"].unique())
-    item_set = set(data["item"].unique())
+def prepare_data(dataset_dir, num_ng=4, total_cols=['user', 'item', "label"]):
+    # Load the train and test datasets
+    data_1 = read_csv(
+        os.path.join(dataset_dir, 'users.dat'),
+        sep="::", header=None, names=['user', 'gender', 'age', 'occupation', 'zipcode'],
+        usecols=[0, 1, 2, 3, 4])
+    data_2 = read_csv(
+        os.path.join(dataset_dir, 'ratings.dat'),
+        sep="::", header=None, names=['user', 'item'],
+        usecols=[0, 1])
+    data_3 = read_csv(
+        os.path.join(dataset_dir, 'movies.dat'),
+        sep="::", header=None, names=['item', 'category'],
+        usecols=[0, 2])
+
+    # calculate numbers of user and item
+    user_set = set(data_2["user"].unique())
+    item_set = set(data_2["item"].unique())
```

Comment:
I think now we need to use the user and item file instead to count for the total users?